complex.go

```diff
@@ -376,14 +376,17 @@ func (t *TypeUnion) Selector(typ Type) int {
 // SplitZNG takes a zng encoding of a value of the receiver's union type and
 // returns the concrete type of the value, its selector, and the value encoding.
 // SplitZNG panics if the selector is invalid.
-func (t *TypeUnion) SplitZNG(zv zcode.Bytes) (Type, int64, zcode.Bytes) {
+func (t *TypeUnion) SplitZNG(zv zcode.Bytes) (Type, zcode.Bytes) {
 	it := zv.Iter()
+	if it.Done() {
+		return t, nil
+	}
```

Comment:
We should test for nil. A non-nil, zero-length `zcode.Bytes` here is a bug and we should panic on that.
```suggestion
	if zv == nil {
		return t, nil
	}
	it := zv.Iter()
```